### PR TITLE
Add reference documentation about indexers and matchers

### LINKED ISSUE #5566: Clarify documentation about indexers and matchers
We could clarify `add_kubernetes_metadata` indexers and matchers functionality: https://www.elastic.co/guide/en/beats/filebeat/6.0/add-kubernetes-metadata.html

Indexers extract pod metadata from kubernetes events and store it in a map in memory
Matchers process beat events and enrich them with metadata from the in-memory map

We should try to avoid indexing related words to avoid confusion with ES indexing, as this is totally unrelated

### PR DESCRIPTION
**Pull request by jsoriano**
Add reference documentation about the available indexers and matchers
in the `add_kubernetes_metadata` processor.
Rephrase some sentences to avoid indexing terminology that could create
confusion with ES indexing.

Fixes #5566.
Closes #14058.

### CODE REVIEW

**Comment by jsoriano** [comment]
> Hey @jsoriano thank you for adding this! How about adding a table of the defaults for each Beat? That was started on #14058, but your PR is more complete.

@ChrsMark ok, good idea to do it here all together. I have added some docs about the defaults mentioned in #14058. Please take another look. Thanks!

File: metricbeat/docs/kubernetes-default-indexers-matchers.asciidoc [hunk 1]
```diff
@@ -0,0 +1,14 @@
+When `add_kubernetes_metadata` is used with {beatname_uc}, it uses the `ip_port`
+indexer and the `fields` matcher with the `metricset.host` field. So events that
+contain a `metricset.host` field are enriched with metadata of the pods that
+exposes the same port in the same IP.
+
+This behaviour can be disabled disabling default indexers and matchers in the
```
Comment (ChrsMark) [inline metricbeat/docs/kubernetes-default-indexers-matchers.asciidoc]: minor but maybe a `by` is needed here:
```suggestion
This behaviour can be disabled by disabling default indexers and matchers in the
```

File: filebeat/docs/kubernetes-default-indexers-matchers.asciidoc [hunk 2]
```diff
@@ -0,0 +1,14 @@
+When `add_kubernetes_metadata` is used with {beatname_uc}, it uses the
+`container` indexer and the `logs_path`. So events whose path in `log.file.path`
+contains a reference to a container ID are enriched with metadata of the pod of
+this container.
+
+This behaviour can be disabled disabling default indexers and matchers in the
```
Comment (ChrsMark) [inline filebeat/docs/kubernetes-default-indexers-matchers.asciidoc]: minor but maybe a `by` is needed here:
```suggestion
This behaviour can be disabled by disabling default indexers and matchers in the
```

Comment (jsoriano) [reply filebeat/docs/kubernetes-default-indexers-matchers.asciidoc]: Better, yes, thanks!

**Pull request merged**